File: ScrollPagerPlugin/bin/Debug/ScrollPagerPlugin.js
// ScrollPagerPlugin.js
(function($){
$.fn.scrollPagerPlugin=function(customOptions){var $0={};$0.pageSize=10;$0.currentPage=1;$0.holder='.listcontainer';$0.viewport='';$0.pageHeight=23;$0.onPageChanged=null;$0.container='#listcontainerdiv';var $1=$.extend({},$0,customOptions);return this.each(function($p1_0,$p1_1){
var $1_0=$(this);var $1_1=1;var $1_2=new PositionInfo();var $1_3=new MouseInfo();var $1_4=0;$1_0.children().each(function($p2_0,$p2_1){
if($p2_0<$1_1*$1.pageSize&&$p2_0>=($1_1-1)*$1.pageSize){$(this).addClass('page'+$1_1);}else{$(this).addClass('page'+($1_1+1));$1_1++;}});var $1_5=$($1.container).height();var $1_6=$1_5;$1_0.children().hide();$('.page'+$1.currentPage).show();if($1_1>1){$1_6=($1_5/$1_1);var $1_12="<UL class=scrollbar sizcache='4' sizset='13'>";for($p1_0=1;$p1_0<=$1_1;$p1_0++){if($p1_0===$1.currentPage){$1_12+="<LI class='currentPage pageItem' sizcache='4' sizset='13'><A class='sliderPage' href='#' rel='"+$p1_0+"'></A>";}else{$1_12+="<LI class='pageNav"+$p1_0+" pageItem' sizcache='4' sizset='14'><A class='sliderPage' href='#' rel='"+$p1_0+"'></A>";}}var $1_13="<LI class='thumb' sizcache='4' sizset='13' style='top:"+($1.currentPage-1)*$1_6+'; height:'+($1_6-3)+"'><A class='sliderThumb' href='#' rel='"+$p1_0+"'></A>";$1_12+=$1_13;$1_12+='</LI></UL>';if(!$1.holder){$1_0.after($1_12);}else{$($1.holder).append($1_12);}$('.pageItem').height($1_6);}var $1_7=$('.scrollbar');var $1_8=$('.thumb');var $1_9=($1_7.height()-$1_8.height());var $1_A=$('.pageItem a');var $1_B=function($p2_0){
var $2_0=$p2_0.attr('rel');$1.onPageChanged($p2_0,new SelectedPageEventArgs(parseInt($2_0)));$1.currentPage=parseInt($2_0);$('li.currentPage').removeClass('currentPage');$p2_0.parent('li').addClass('currentPage');$1_0.children().hide();$1_0.find('.page'+$2_0).show();};var $1_C=function($p2_0){
var $2_0=$($p2_0.currentTarget);$1_B($2_0);};$1_A.live('click',$1_C);var $1_D=function($p2_0){
};var $1_E=function($p2_0){
var $2_0=Math.max(0,($1_2.start+($p2_0.pageY-$1_3.start)));$1_2.now=($2_0>$1_9)?$1_9:$2_0;$1_4=Math.round($1_2.now/$1_8.height());$1_8.css('top',$1_2.now.toString());};var $1_F=null;$1_F=function($p2_0){
$(document).unbind('mousemove',$1_E);$(document).unbind('mouseup',$1_F);$1_8.die('mouseup',$1_F);$1_B($($1_A[$1_4]));};var $1_10=function($p2_0){
$1_3.start=$p2_0.pageY;var $2_0=$1_8.css('top');$1_2.start=($2_0==='auto')?0:parseInt($2_0);$(document).bind('mousemove',$1_E);$(document).bind('mouseup',$1_F);$1_8.live('mouseup',$1_F);};var $1_11=function(){
$1_8.live('mousedown',$1_10);$1_7.live('mouseup',$1_E);};$1_11();});}
window.MouseInfo=function(){}
MouseInfo.prototype={start:0,end:0}
window.PositionInfo=function(){}
PositionInfo.prototype={start:0,now:0}
window.SelectedPageEventArgs=function(selectedPageNumber){SelectedPageEventArgs.initializeBase(this);this.selectedPage=selectedPageNumber;}
SelectedPageEventArgs.prototype={selectedPage:0}
MouseInfo.registerClass('MouseInfo');PositionInfo.registerClass('PositionInfo');SelectedPageEventArgs.registerClass('SelectedPageEventArgs',ss.EventArgs);})(jQuery);// This script was generated using Script# v0.7.0.0

File: ScrollPagerPlugin/ScrollPagerPlugin.cs
// ScrollPagerPlugin.cs
//

using System;
using System.Html;
using System.Runtime.CompilerServices;
using jQueryApi;

[Mixin("$.fn")]
public static class ScrollPagerPluginPlugin
{
	public static jQueryObject ScrollPagerPlugin(ScrollPagerPluginOptions customOptions)
	{
		ScrollPagerPluginOptions defaultOptions =new ScrollPagerPluginOptions();
		defaultOptions.pageSize = 10;
		defaultOptions.currentPage = 1;
		defaultOptions.holder = ".listcontainer";
		defaultOptions.viewport = "";
		defaultOptions.pageHeight = 23;
		defaultOptions.onPageChanged = null;
		defaultOptions.container = "#listcontainerdiv";

		ScrollPagerPluginOptions options =jQuery.ExtendObject<ScrollPagerPluginOptions>(new ScrollPagerPluginOptions(), defaultOptions, customOptions);

		return jQuery.Current.Each(delegate(int i, Element element)
		                           	{
		                           		jQueryObject selector = jQuery.This;
										int pageCounter = 1;
										PositionInfo iPosition = new PositionInfo();
										MouseInfo iMouse = new MouseInfo();
		                           		Number candidatePageIndex = 0;

										//this goes through every item in the 
		                           		selector
											.Children()
											.Each(delegate(int ic, Element elc)
		                           		                         	{
																		if (ic < pageCounter * options.pageSize && ic >= (pageCounter - 1) * options.pageSize)
																		{
																			jQuery.This.AddClass("page" + pageCounter);
																		}
																		else
																		{
																			jQuery.This.AddClass("page" + (pageCounter+1));
																			pageCounter++; 
																		}
		                           		                         	});

										//set and default the slider item height
										int contHeight = jQuery.Select(options.container).GetHeight();
										int sliderItemHeight = contHeight;

										//show/hide the appropriate regions 
										selector.Children().Hide();
										jQuery.Select(".page" + options.currentPage).Show();

										//more than one page? 
										if (pageCounter > 1)
										{
											//calculate the slider item height
											sliderItemHeight = (contHeight / pageCounter);

											//Build pager navigation
											string pageNav = "<UL class=scrollbar sizcache='4' sizset='13'>";
											for (i = 1; i <= pageCounter; i++)
											{
												if (i == options.currentPage)
												{
													pageNav += "<LI class='currentPage pageItem' sizcache='4' sizset='13'><A class='sliderPage' href='#' rel='" + i + "'></A>";
												}
												else
												{
													pageNav += "<LI class='pageNav" + i + " pageItem' sizcache='4' sizset='14'><A class='sliderPage' href='#' rel='" + i + "'></A>";
												}

											}

											//Create slider item 
											string sliderItem = "<LI class='thumb' sizcache='4' sizset='13' style='top:" + (options.currentPage - 1) * sliderItemHeight + "; height:" + (sliderItemHeight - 3) + "'><A class='sliderThumb' href='#' rel='" + i + "'></A>";
											pageNav += sliderItem;
											pageNav += "</LI></UL>";

											if (options.holder == "")
											{
												selector.After(pageNav);
											}
											else
											{
												jQuery.Select(options.holder).Append(pageNav);
											}

											//Apply the slider item height 
											jQuery.Select(".pageItem").Height(sliderItemHeight);

										}

										jQueryObject oTrack = jQuery.Select(".scrollbar");
										jQueryObject oThumb = jQuery.Select(".thumb");
										//jQueryObject oViewPort = jQuery.Select(options.viewport);
										Number maxPos = (oTrack.GetHeight() - oThumb.GetHeight());
										jQueryObject pageItemCollection = jQuery.Select(".pageItem a");

		                           		Action<jQueryObject> selectPageItem = delegate(jQueryObject pageItem)
		                           		                                      	{
																					string clickedLink = pageItem.GetAttribute("rel");
																					options.onPageChanged.Invoke(pageItem, new SelectedPageEventArgs(int.Parse(clickedLink)));

																					options.currentPage = int.Parse(clickedLink);
																					//remove current current (!) page
																					jQuery.Select("li.currentPage").RemoveClass("currentPage");
																					//Add current page highlighting
																					pageItem.Parent("li").AddClass("currentPage");
																					//hide and show relevant links				
																					selector.Children().Hide();
																					selector.Find(".page" + clickedLink).Show();
		                           		                                      	};

										//Action<jQueryObject> selectPageItem = delegate(jQueryObject pageItem)
										jQueryEventHandler selectPageItemHandler = delegate(jQueryEvent pageItemClickedEvent)
										{
											//grab the REL attribute 
											jQueryObject pageItem = jQuery.FromElement(pageItemClickedEvent.CurrentTarget);
											selectPageItem(pageItem);
										};

										
										//pager navigation behaviour
		                           		pageItemCollection.Live("click", selectPageItemHandler);

										jQueryEventHandler wheel = delegate(jQueryEvent oEvent)
		                           		                           	{
		                           		                           	};

										jQueryEventHandler drag = delegate(jQueryEvent oEvent)
										{
											Number candidatePos = Math.Max(0, (iPosition.Start + ((oEvent.PageY) - iMouse.Start)));
											iPosition.Now = (candidatePos > maxPos) ? maxPos : candidatePos;
											candidatePageIndex = Math.Round(iPosition.Now / oThumb.GetHeight());
											oThumb.CSS("top", iPosition.Now.ToString()); ;
										};

										jQueryEventHandler end = null; 
										end = delegate(jQueryEvent oEvent)
										{
											jQuery.Document.Unbind("mousemove", drag);
											jQuery.Document.Unbind("mouseup", end);
											oThumb.Die("mouseup", end);
											selectPageItem(jQuery.FromElement(pageItemCollection[candidatePageIndex]));
										};

										jQueryEventHandler start = delegate(jQueryEvent oEvent)
										{
											iMouse.Start = oEvent.PageY;
											string oThumbDir = oThumb.GetCSS("top");
											iPosition.Start = (oThumbDir == "auto") ? 0 : int.Parse(oThumbDir);
											jQuery.Document.Bind("mousemove", drag);
											jQuery.Document.Bind("mouseup", end);
											oThumb.Live("mouseup", end);
										};

										Action setEvents = delegate()
										{
											oThumb.Live("mousedown", start);
											oTrack.Live("mouseup", drag);
											//if (options.scroll != null)
											//{
											//    oViewPort[0].AddEventListener("DOMMouseScroll", wheel, false);
											//    oViewPort[0].AddEventListener("mousewheel", wheel, false);
											//}
											//else if (options.scroll != null)
											//{
											//    oViewPort[0].OnMouseWheel = wheel;
											//}
										};

										setEvents.Invoke();

		                           	});
	}
}

public class MouseInfo
{
	public Number Start = 0;
	public Number End = 0;
}


public class PositionInfo
{
	public Number Start = 0;
	public Number Now = 0;
}

public class SelectedPageEventArgs : EventArgs
{
	public int selectedPage = 0;
	public SelectedPageEventArgs(int selectedPageNumber)
	{
		selectedPage = selectedPageNumber;
	}
}

[Imported]
[ScriptName("Object")]
public sealed class ScrollPagerPluginOptions
{
	// TODO: Replace with fields corresponding to options for the plugin
	public int pageSize = 10;
	public int currentPage = 1;
	public int totalRecords = 100; 
	public string holder = "";
	public string viewport = "";
	public string container = "#listcontainerdiv";
	public int pageHeight = 0;
	public EventHandler<SelectedPageEventArgs> onPageChanged; //delegate(object o, SelectedPageEventArgs e)
	                                                          //  	{
																//		Window.Top.Status = "Selected Page: " + e.selectedPage.ToString();
	                                                           // 	};
	public object scroll; 
	


	public ScrollPagerPluginOptions()
	{
	}

	public ScrollPagerPluginOptions(params object[] nameValuePairs)
	{
	}
}

#region Script# Support
[Imported]
public sealed class ScrollPagerPluginObject : jQueryObject
{
	public jQueryObject ScrollPagerPlugin()
	{
		return null;
	}

	public jQueryObject ScrollPagerPlugin(ScrollPagerPluginOptions options)
	{
		return null;
	}
}
#endregion
